tests/components/zha/test_registries.py

```diff
@@ -197,7 +208,49 @@ def test_registry_matching(rule, matched, zha_device, channels):
         ),
     ],
 )
-def test_registry_loose_matching(rule, matched, zha_device, channels):
+def test_registry_loose_matching(rule, matched, channels):
     """Test loose rule matching."""
-    reg = registries.ZHAEntityRegistry()
-    assert reg._loose_matched(zha_device, channels, rule) is matched
+    assert rule.loose_matched(MANUFACTURER, MODEL, channels) is matched
+
+
+def test_match_rule_claim_channels_color(channel):
+    """Test channel claiming."""
+    ch_color = channel("color", 0x300)
+    ch_level = channel("level", 8)
+    ch_onoff = channel("on_off", 6)
+
+    rule = registries.MatchRule(channel_names="on_off", aux_channels={"color", "level"})
+    claimed = rule.claim_channels([ch_color, ch_level, ch_onoff])
+    assert {"color", "level", "on_off"} == set([ch.name for ch in claimed])
+
+
+@pytest.mark.parametrize(
+    "rule, match",
+    [
+        (registries.MatchRule(channel_names={"level"}), {"level"}),
+        (registries.MatchRule(channel_names={"level", "no match"}), {"level"}),
+        (registries.MatchRule(channel_names={"on_off"}), {"on_off"}),
+        (registries.MatchRule(generic_ids="channel_0x0000"), {"basic"}),
+        (
+            registries.MatchRule(channel_names="level", generic_ids="channel_0x0000"),
+            {"basic", "level"},
+        ),
+        (registries.MatchRule(channel_names={"level", "power"}), {"level", "power"}),
+        (
+            registries.MatchRule(
+                channel_names={"level", "on_off"}, aux_channels={"basic", "power"}
+            ),
+            {"basic", "level", "on_off", "power"},
+        ),
+        (registries.MatchRule(channel_names={"color"}), set()),
+    ],
+)
+def test_match_rule_claim_channels(rule, match, channel, channels):
+    """Test channel claiming."""
+    ch_basic = channel("basic", 0)
+    channels.append(ch_basic)
+    ch_power = channel("power", 1)
+    channels.append(ch_power)
+
+    claimed = rule.claim_channels(channels)
+    assert match == set([ch.name for ch in claimed])
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5051600)